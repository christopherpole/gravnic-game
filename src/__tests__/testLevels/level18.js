import { MOVE_LEFT, ENTITIES, levelIsComplete, makeMoves } from '../../index';

describe('Level 18', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR, id: 2 },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#800080', id: 1 },
        },
        { staticEntity: null, movableEntity: null },
      ],
      [
        {
          staticEntity: { entityId: ENTITIES.FLOOR, id: 4 },
          movableEntity: { entityId: ENTITIES.SMART_BOMB, id: 3 },
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 5 }, movableEntity: null },
        {
          staticEntity: {
            entityId: ENTITIES.COLOR_CHANGER,
            targetEntity: { entityId: ENTITIES.RAINBOW_BLOCK },
            id: 6,
          },
          movableEntity: null,
        },
        {
          staticEntity: { entityId: ENTITIES.FLOOR, id: 8 },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 7 },
        },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_LEFT]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
