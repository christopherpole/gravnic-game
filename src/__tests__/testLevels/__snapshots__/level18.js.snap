// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 18 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "RAINBOW_BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "COLOR_CHANGER",
            "id": 6,
            "targetEntity": Object {
              "entityId": "RAINBOW_BLOCK",
            },
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "RAINBOW_BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "COLOR_CHANGER",
            "id": 6,
            "targetEntity": Object {
              "entityId": "RAINBOW_BLOCK",
            },
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "fading": true,
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "fading": true,
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "RAINBOW_BLOCK",
            "fading": true,
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "COLOR_CHANGER",
            "id": 6,
            "targetEntity": Object {
              "entityId": "RAINBOW_BLOCK",
            },
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "COLOR_CHANGER",
            "id": 6,
            "targetEntity": Object {
              "entityId": "RAINBOW_BLOCK",
            },
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
    ],
  ],
]
`;
